src/app/shared/pipes/orderBy/order-by.pipe.ts

```diff
@@ -0,0 +1,19 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { Course } from '@modules/courses/types/course';
+
+@Pipe({
+  name: 'orderBy'
+})
+export class OrderByPipe implements PipeTransform {
+
+  transform(courses: Course[], direction: 'asc' | 'desc' = 'asc'): Course[] {
+    return courses.sort((prev, next) => {
+      const prevTime = new Date(prev.creationDate);
```

Comment:
It will be better if you will make this pipe more saleable. I would suggest to move sork key name to argument.  `transform(courses: Course[], direction: 'asc' | 'desc' = 'asc', key = 'creationDate')`

Reply:
in my case, creationDate is a specific key which is date. No other keys in course will refer to date. Or i should refactor whole pipe, in order to create some